webproject/src/main/java/com/google/sps/dataManagers/UserRoomManager.java

```diff
@@ -0,0 +1,70 @@
+package com.google.sps.dataManagers;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.LinkedBlockingDeque;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import com.google.firebase.database.ChildEventListener;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.ValueEventListener;
+import com.google.sps.data.UserRoom;
+import com.google.sps.util.FirebaseUtil;
+
+@Singleton
+public class UserRoomManager {
+    private FirebaseUtil firebaseUtil;
+
+    @Inject
+    public UserRoomManager(FirebaseUtil firebaseUtil) {
+        this.firebaseUtil = firebaseUtil;
+    }
+
+    public void addUserRoom(String userEmail, String roomId) {
+        UserRoom userRoom = UserRoom.newBuilder().setUserEmail(userEmail).setRoomId(roomId).setUserEmailRoom().build();
+
+        firebaseUtil.getUserRoomReference().push()
+                .setValue(userRoom, (databaseError, databaseReference) -> {
+                    if (databaseError != null) {
+                        System.out.println("Data could not be saved " + databaseError.getMessage());
+                    } else {
+                        System.out.println("Data saved successfully.");
+                    }
+                });
+    }
+
+    public boolean hasUserJoinedRoom(String userEmail, String roomId) {
+        String userEmailRoom = userEmail + "_" + roomId;
+        final BlockingQueue queue = new LinkedBlockingDeque(1);
```

Comment:
if you are using BlockingQueue, CountdownLatch is not necessary anymore
you can use queue.poll() with timeout of 30 seconds instead of done.await()

Reply:
done